src/models/currentUser.js

```diff
@@ -37,17 +37,22 @@ export const isTester = user =>
     {id: '1972', slug: 'testmetalab'}
   ], membership(user))
 
-export const canEditPost = (currentUser, post) => {
-  return same('id', currentUser, post.user) ||
-    some(post.communities.map(c => c.id ? c.id : c), canModerate(currentUser)) ||
-    isAdmin(currentUser)
+export const canEditPost = (currentUser, permissionsPost) => {
+  if (currentUser && permissionsPost) {
+    return same('id', currentUser, permissionsPost.user) ||
+      some(permissionsPost.communities.map(c => c.id ? c.id : c), canModerate(currentUser)) ||
+      isAdmin(currentUser)
+  }
 }
 
-export const canComment = (currentUser, post) =>
-  currentUser && includes(post.follower_ids, currentUser.id) ||
-    !isEmpty(intersection(
-      map('community_id', get('memberships', currentUser)),
-      post.community_ids))
+export const canCommentOnPost = (currentUser, permissionsPost) => {
+  if (currentUser && permissionsPost) {
```

Comment:
could have just tacked `permissionsPost &&` onto the front of the chain here and put parentheses around the || part